### PR TITLE
Adjust whitespace and comments on Dockerfile stages

### PR DESCRIPTION
**Pull request by rubys**
### Motivation / Background

Provide visual separation and comments for the docker build stages, per suggestion from @dhh

### Detail

This commit consists entirely of whitespace and comments.

### Additional information

Note that the first comment has a function:

https://docs.docker.com/build/buildkit/dockerfile-frontend/

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`

No tests or changelog are needed for this change.

### CODE REVIEW

**Comment by dhh** [comment]
Are we using any syntax currently that benefits from that first line? It feels like warty to me. I've looked at a lot of Dockerfiles, I've also not seen it before.

File: railties/lib/rails/generators/rails/app/templates/Dockerfile.tt [hunk 1]
```diff
@@ -49,6 +51,7 @@ RUN SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile
 
 <% end -%>
 
+# Final stage for app image
 FROM base
 
 # copy built artifacts: libraries, gems, application
```
Comment (dhh) [inline railties/lib/rails/generators/rails/app/templates/Dockerfile.tt:57]: Should capitalize Copy to fit with the rest of the comment style.

Comment (rubys) [reply railties/lib/rails/generators/rails/app/templates/Dockerfile.tt]: Fixed.

**Comment by rubys** [comment]
> Are we using any syntax currently that benefits from that first line? It feels like warty to me. I've looked at a lot of Dockerfiles, I've also not seen it before.

I know that you are old enough to remember when DOCTYPES in HTML were rare.  And yet here we are and application.html.erb and each publc/nnn.html file now has one.

But to answer your question: no, we aren't using any syntax now that requires this.  I don't recall whether adding buildkit caching requires it - I'll have to do some testing.  I also know that there will be cases where it is convenient to recommend that the user add small heredoc into the file, which they won't be able to do unless this comment is present.  But I'll remove it for now and pursue it as a separate pull request.

**Comment by rubys** [comment]
Did my research.  Yes, buildkit caching requires it, and it is recommended.  WIll follow up on https://discuss.rubyonrails.org/c/rubyonrails-core/5

**Comment by rubys** [comment]
Posted: https://discuss.rubyonrails.org/t/rails-dockerfile-futures/82091

**Pull request merged**